src/DateUtils.ts

```diff
@@ -18,90 +18,66 @@ limitations under the License.
 
 import { Optional } from "matrix-events-sdk";
 
-import { _t } from "./languageHandler";
+import { _t, getUserLanguage } from "./languageHandler";
 
-function getDaysArray(): string[] {
-    return [_t("Sun"), _t("Mon"), _t("Tue"), _t("Wed"), _t("Thu"), _t("Fri"), _t("Sat")];
-}
-
-function getMonthsArray(): string[] {
-    return [
-        _t("Jan"),
-        _t("Feb"),
-        _t("Mar"),
-        _t("Apr"),
-        _t("May"),
-        _t("Jun"),
-        _t("Jul"),
-        _t("Aug"),
-        _t("Sep"),
-        _t("Oct"),
-        _t("Nov"),
-        _t("Dec"),
-    ];
-}
+const MILLIS_IN_DAY = 86400000;
 
-function pad(n: number): string {
-    return (n < 10 ? "0" : "") + n;
+export function getDaysArray(weekday: Intl.DateTimeFormatOptions["weekday"] = "short"): string[] {
+    const now = new Date();
+    const { format } = new Intl.DateTimeFormat(getUserLanguage(), { weekday });
+    return [...Array(7).keys()].map((day) => format(now.getTime() - (now.getDay() - day) * MILLIS_IN_DAY));
```

Comment:
I feel like using the current time in this calculation is... kinda hard to grok.

An alternative solution: hardcode the timestamp of a time known to be a Sunday, and work from there:

```suggestion
    const sunday = 1672574400000; // 2023-01-01 12:00 UTC
    const { format } = new Intl.DateTimeFormat(getUserLanguage(), { weekday, timeZone: "UTC" });
    return [...Array(7).keys()].map((day) => format(sunday + day * MILLIS_IN_DAY));
```